### DISCUSSION TITLE
Commercial use of images generated using Fooocus

### DISCUSSION THREAD

**Discussion by Rectiras** [lllyasviel/Fooocus#1957] [Q&A] [answered]
I have been in contact with a company and I have informed them of Fooocus which they could use for their products. Instead they offered me commissions on the sales they make with the products using the images generated by me using Fooocus. Is this legal? Do I have the rights to sell my images to the company for commercial use?

**Comment by Njenn** [comment]
According to the current legal landscape, most companies currently would want to (knowingly) stay away from AI generated art because it isn't copyrightable. So I could totally copy/reuse/resell the AI images that you sold and there would currently be no legal repercussions.

**Reply by Rectiras** [reply]
Thanks a lot for your explanation

**Comment by docppp** [comment]
Also, you have no knowledge if image you get is truly generated and not "stolen" from some forgotten deviantart page. I have already seen that ai generates image with signature of legit artist and maybe this particular image was a little bit different (so you may feel like covering it in photoshop is a good idea), but you never know. Google "nyt sues openai" if you want some recent examples ;)

**Reply by Rectiras** [reply]
Thank you so much, I will be on the look out for such details.

**Comment by mashb1t** [comment, marked as answer]
Generally speaking you legally have the rights to sell your images, as the SDXL license permits this (see https://huggingface.co/stabilityai/stable-diffusion-xl-base-1.0/blob/main/LICENSE.md) - which can be modified per fine-tuned model, so this only applies to SDXL base but not necessarily for models based on it:

> each Contributor hereby grants to You a perpetual, worldwide, non-exclusive, no-charge, royalty-free, irrevocable (except as stated in this paragraph) patent license to make, have made, use, **offer to sell, sell**, import, and otherwise transfer the Model and the Complementary Material, where such license applies only to those patent claims licensable by such Contributor that are necessarily infringed by their Contribution(s) alone or by combination of their Contribution(s) with the Model to which such Contribution(s) was submitted.

But, as pointed out previously, the used dataset might have included also a bit of copyrighted or claimed material. This is a small percentage though AFAIK, but i'm no lawyer.

> Dataset 
SDXL used an internal dataset for pretraining and fine-tuning the model. The dataset consists of 1.8 million images from the ImageNet dataset and 1.2 million images from the OpenImages dataset. The images were resized to 256 x 256 pixels and augmented with random crops, flips, and rotations. The authors also used a subset of the COCO dataset for evaluation. 
> The ImageNet dataset is a large-scale dataset of natural images that is widely used for computer vision tasks. It consists of over 1.2 million images with 1000 object categories. The OpenImages dataset is another large-scale dataset of natural images that consists of over 9 million images with 600 object categories. The COCO dataset is a popular dataset for object detection and segmentation tasks that consists of over 330,000 images with 80 object categories. 

Source: https://clarifai.com/stability-ai/stable-diffusion-2/models/stable-diffusion-xl

You can find more in the official SDXL paper, https://huggingface.co/papers/2307.01952

**Reply by Rectiras** [reply]
Thanks a lot for your detailed explanations.

**Comment by mashb1t** [comment]
@Rectiras fyi

> Midjourney's attention formulation of image prompt (non-official reimplementation)
> Written by Lvmin Zhang at Stanford University, 2023 Dec
> For non-commercial use only - if you use this in commercial project then
> probably it has some intellectual property issues.
> Contact lvminzhang@acm.org if you are not sure.

https://github.com/lllyasviel/Fooocus/blob/main/extras/ip_adapter.py#L228-L232

**Comment by blackKravitz** [comment]
Hi. I have a similar question. Can I use focusus to create AI avatars for commercial purposes? It is free?

**Reply by mashb1t** [reply]
see
- https://github.com/lllyasviel/Fooocus/discussions/1957#discussioncomment-8175629
- https://github.com/lllyasviel/Fooocus/discussions/1957#discussioncomment-8283320 
- https://github.com/lllyasviel/Fooocus/blob/main/LICENSE

**Reply by blackKravitz** [reply]
thank you so much! :) we are trying to create something but we don't know the limit and so i asked 4 this :)

**Reply by blackKravitz** [reply]
it's not clear to me, i read:   Permissions
Commercial use
Modification
Distribution
Patent use
Private use
Limitations
Liability
Warranty
Conditions
License and copyright notice
State changes
Disclose source
Same license. So, it's yes or not? could you tell me, please?

**Reply by mashb1t** [reply]
> Using the Licensed Code
> The GPL v3 license permits users of the code to:
> 
> - Use the code for commercial purposes: Like GPL v2, GPL v3 imposes no conditions on the internal use of the software.
> - Change the code: Users can change or rework the code, but if they distribute these changes/modifications in binary form, they’re also required to release these updates in source code form under the GPL v3 license.
> - Distribute copies or modifications of the code: As long as these modifications are also released under the GPL v3 license, they can be distributed to others.
> - Place warranty: Distributors of the original code can offer their own warranty on the licensed software.
> 
> **Like its predecessor, GPL v3 does not allow users to sublicense the code. In other words, you cannot rework, alter, or add to the code, then close those changes off to the public. The “open source-ness” of the original code follows any update or addition.**

https://fossa.com/blog/open-source-software-licenses-101-gpl-v3/

I am no lawyer, but as far as i understand you can use Fooocus without modifications commercially as part of your toolchain, but you can't distribute the code without making also your code available open-source and under GPLv3. Please take this with a grain of salt and get legal advice if necessary.

**Reply by firugmia44** [reply]
Where Will I get **the Licensed Code**??

**Comment by firugmia44** [comment]
Is there any copyright issue of fooocusai? Can I use commercially fooocusai generated images on my youtube Channel?
